core/rest-client/src/http_client.rs

```diff
@@ -122,6 +128,7 @@ impl SendHttpRequest for HttpClient {
 	fn send_request<U, T>(
 		&self,
 		base_url: Url,
+		root_certificate: Option<String>,
```

Comment:
Do we really need the `authenticated_connection` flag? Don't we have the same behavior with this parameter? If the `root_certificate` is `Some`, we call the version with an explicit certificate. When it's `None`, we call the version with the default root certificates.

Reply:
So we have a real use case where the root certificate parameter is `None`, but `authenticated_connection` is `true`?

Reply:
Such a use case is an error. And we want to detect such errors. In use cases, where we want to restrict TLS connections to the condition that a valid server root certificate is provided, the user could bypass this by not providing any certificate.